UPGRADE.md

```diff
@@ -8,6 +8,14 @@ you are using a URL-style DSN and have a username, password or database name con
 percentage sign, you need to update your DSN. If your password is, say, ``foo%foo``, it
 should be encoded as ``foo%25foo``.
 
+## MINOR BC BREAK: Native JSON type support in MySQL 5.7 platform
+
+MySQL 5.7 ships with native JSON type support which now is adopted by DBAL.
+If you are using MySQL 5.7 and have columns with ``json_array`` type defined in your database,
+utilizing DBAL's schema comparator will cause a schema diff, where the previously used native type
+``LONGTEXT`` will be changed to ``JSON``.
```

Comment:
@rh389 good point. Considering that, I can remove the upgrade notice again I suppose unless there are other reasons why BC could not be maintained here. Currently I can't think of any.